nexus/src/app/background/tasks/v2p_mappings.rs

```diff
@@ -44,23 +42,19 @@ impl BackgroundTask for V2PManager {
 
             // Get sleds
             // we only care about sleds that are active && inservice
-            let sleds = match self.datastore.sled_list_all_batched(opctx, SledFilter::InService).await
+            let sleds = match self.datastore.sled_list_all_batched(opctx, SledFilter::V2PMapping).await
```

Comment:
I've basically cribbed the same logic from this block to select the sleds we talk to in the VPC route background task:

https://github.com/oxidecomputer/omicron/blob/97fe552507bf5cdfaaa85a3256e337df5b5b05c4/nexus/src/app/background/tasks/vpc_routes.rs#L60-L88

Is this fix something we should apply there, too?

Reply:
It could cause issues if we want these sleds to receive updates / be managed even after they are marked as `non_provisionable`, as in this issue:
https://github.com/oxidecomputer/omicron/issues/5872

Does that scenario apply here?

Reply:
I think it does; my understanding of 'non-provisionable' is that we're not placing new instances on such a sled, but existing ones should still function as intended? In that case we should be making sure that those sleds have up-to-date routing tables for their instances' OPTE ports while they still exist.

Reply:
Ok, so maybe we need to rename it from `SledFilter::V2PMapping` to `SledFilter::VpcConfig` or similar and apply it to both RPWs?

Reply:
I think that would work!

Reply:
I'd like to make this less error-prone! The docs on `SledFilter` have this note:

```
/// The meaning of a particular filter should not be overloaded -- each time a
/// new use case wants to make a decision based on the zone disposition, a new
/// variant should be added to this enum.
```

but I imagine it's easy to miss _and_ `InService` seems general to a fault, given that guidance. I'm inclined to try removing the `InService` variant altogether and updating all its users with more specific variant names, but am open to other suggestions!

Reply:
Also FWIW, I believe `SledFilter::InService` actually does do the right thing here - it includes sleds that are not provisionable. #5872 was caused by the raw DB query inspecting the sled policy/state columns directly and not going through the `SledFilter` machinery at all. That said, I stand by the previous point that I think `InService` is still too general and will still try the renaming exercise.

Reply:
Looks good, thank you!